### PR TITLE
[Repo] basic-service -> example-service

### PR DESCRIPTION
**Pull request by arturgontijo**
Replacing `example-service` with `basic-service`.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,81 +12,108 @@ Simple image classification service compatible with the SingularityNET daemon
...
+tutorial to publish this service and set the `snetd.config.json` file.
+
+Then, be sure that you have the `snetd` in your PATH.
+
+Create the `snetd.config.json` file using this template:
+
+```
 {
-  "passthrough_enabled": true
+   "DAEMON_LISTENING_PORT": DAEMON_PORT,
```
Comment (vsbogd) [inline README.md]: don't needed
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -12,81 +12,108 @@ Simple image classification service compatible with the SingularityNET daemon
...
+   "DAEMON_END_POINT": "DAEMON_HOST:DAEMON_PORT",
```
Comment (vsbogd) [inline README.md]: It is critical to have same format of endpoint in published service metadata and here.
In [howToPublishService tutorial](https://github.com/singnet/wiki/tree/master/tutorials/howToPublishService) which is referenced above `http://54.203.198.53:7000` format is used same format should be used here.
```suggestion
   "DAEMON_END_POINT": "http://DAEMON_HOST:DAEMON_PORT",
```

File: README.md [hunk 3]
```diff
@@ -12,81 +12,108 @@ Simple image classification service compatible with the SingularityNET daemon
...
+   "ETHEREUM_JSON_RPC_ENDPOINT": "https://kovan.infura.io",
+   "IPFS_END_POINT": "http://ipfs.singularitynet.io:80",
+   "REGISTRY_ADDRESS_KEY": "0x2e4b2f2b72402b9b2d6a7851e37c856c329afe38",
+   "PASSTHROUGH_ENABLED": true,
+   "PASSTHROUGH_ENDPOINT": "SERVICE_GRPC_HOST:SERVICE_GRPC_PORT",
```
Comment (vsbogd) [inline README.md]: PASSTHROUGH_ENDPOINT should be in URL format:
```suggestion
   "PASSTHROUGH_ENDPOINT": "http://SERVICE_GRPC_HOST:SERVICE_GRPC_PORT",  
```

File: README.md (hunk 2 above)
Comment (arturgontijo) [reply README.md]: Right after the template I gave an example on how to replace each tag.
`DAEMON_HOST:DAEMON_PORT: http://54.203.198.53:7000`
And
`SERVICE_GRPC_HOST:SERVICE_GRPC_PORT: http://localhost:7003`

Do you think that's better add the scheme|protocol on template too?

Comment (vsbogd) [reply README.md]: I think it is confusing to have `:` in template name. When I see 'daemon_host:daemon_port' notation I always replace host by actual host and port by actual port in my mind and see 'localhost:8080'.

If it is a name of template then I would suggest using `daemon_host_port` or `<daemon_host_port>` notation

**Comment by vsbogd** [comment]
Looks good to me

**Comment by astroseger** [comment]
1. I would propose to not specify the endpoint (http://54.203.198.53:7000) in README.md. We could replace it with <your_ip>:<your_port> or something like this.
2. "docker build -t snet_example_service https://github.com/arturgontijo/example-service.git#master" failed
with "sh buildproto.sh' returned a non-zero code: 127"

File: service/common.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+# From gRPC docs:
+# Because start() does not block you may need to sleep-loop if there is nothing
+# else for your code to do while serving.
+def main_loop(grpc_handler, args):
+    server = grpc_handler(port=args.grpc_port)
+    server.start()
+    try:
+        while True:
+            time.sleep(0.1)
```
Comment (astroseger) [inline service/common.py:27]: (It is very minor comment. ). Why we have time.sleep(0.1)? We could put time.sleep(1000), or at least classical time.sleep(1) (server loop is in another thread...).

**Comment by arturgontijo** [comment]
> 1. I would propose to not specify the endpoint (http://54.203.198.53:7000) in README.md. We could replace it with <your_ip>:<your_port> or something like this.
> 2. "docker build -t snet_example_service https://github.com/arturgontijo/example-service.git#master" failed
>    with "sh buildproto.sh' returned a non-zero code: 127"

1 - My idea was to specify a "real" example, maybe replace it with `http://localhost:7000`?
2 - Yeah, the `buildproto.sh` isn't at repo right now, so the `command not found (127)`.

File: service/common.py (hunk 4 above)
Comment (arturgontijo) [reply service/common.py:27]: Ok.

**Comment by arturgontijo** [comment]
Moved to https://github.com/singnet/example-service/pull/1

**Pull request closed without merging**